nityam_kumar/PostgresAssignment/src/services/mark.ts

```diff
@@ -0,0 +1,205 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import db from "../db";
+
+import AppError from "../utils/appError";
+
+interface IMark {
+  student_id: string;
+  subject_id: string;
+  marks: number;
+  teacher_id: string;
+  class_number: number;
+}
+
+export const fetchMarks = async (student_id: string) => {
+  const data = await db.query(
+    "select s.fname,s.lname,s.cl_no,m1.marks from marks as m1,student as s where m1.st_id=$1 and s.st_id= m1.st_id order by s.fname",
+    [student_id.trim()]
+  );
+  if (data.rows.length === 0) {
```

Comment:
why should fetchMarks worry about no records found ??? should service layer throw error if data is not found ?

Reply:
if student with this id does not exist then it will return error that student_id not found

Reply:
moved all thrown errors if data is not found to controllers